src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -83,6 +91,145 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
+        final int numNewAltAlleles = allelesToUse.size() - 1;
+        final List<Integer> sacIndicesToUse = numOriginalAltAlleles == numNewAltAlleles ? null : determineSACIndicesToUse(originalAlleles, allelesToUse);
+
+        return makeNewSACs(g, sacIndicesToUse);
+    }
+
+    /**
+     * Make a new SAC array from the a subset of the genotype's original SAC
+     *
+     * @param g               the genotype
+     * @param sacIndicesToUse the indices in the SAC to use given the allelesToUse (@see #determineSACIndicesToUse())
+     * @return subset of SACs from the original genotype, the original SACs if sacIndicesToUse is null
+     */
+    private static int[] makeNewSACs(final Genotype g, final List<Integer> sacIndicesToUse) {
+
+        final int[] oldSACs  = getSACs(g);
+
+        if (sacIndicesToUse == null) {
+            return oldSACs;
+        } else {
+            final int[] newSACs = new int[sacIndicesToUse.size()];
+            int newIndex = 0;
+            for (final int oldIndex : sacIndicesToUse) {
+                newSACs[newIndex++] = oldSACs[oldIndex];
+            }
+            return newSACs;
+        }
+    }
+
+    /**
+     * Get the genotype SACs
+     *
+     * @param g the genotype
+     * @return an arrays of SACs
+     * @throws GATKException if the type of the SACs is unexpected
+     */
+    private static int[] getSACs(final Genotype g) {
+
+        if ( !g.hasExtendedAttribute(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY) )
+            throw new IllegalArgumentException("Genotype must have SAC");
+
+        if ( g.getExtendedAttributes().get(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY).getClass().equals(String.class) ) {
+            final String SACsString = (String) g.getExtendedAttributes().get(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY);
+            String[] stringSACs = SACsString.split(",");
+            final int[] intSACs = new int[stringSACs.length];
+            int i = 0;
+            for (String sac : stringSACs)
+                intSACs[i++] = Integer.parseInt(sac);
+
+            return intSACs;
+        }
+        else if ( g.getExtendedAttributes().get(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY).getClass().equals(int[].class) )
+            return (int[]) g.getExtendedAttributes().get(GATKVCFConstants.STRAND_COUNT_BY_SAMPLE_KEY);
+        else
+            throw new GATKException("Unexpected SAC type");
+    }
+
+    /**
+     * Find the strand allele count indices to use for a selected set of alleles
+     *
+     * @param originalAlleles the original alleles
+     * @param allelesToUse the subset of alleles to use
+     * @return a list of SAC indices to use or null if none
+     */
+    private static List<Integer> determineSACIndicesToUse(final List<Allele> originalAlleles, final List<Allele> allelesToUse) {
+
+        // the bitset representing the allele indices we want to keep
+        final BitSet alleleIndicesToUse = getAlleleIndexBitset(originalAlleles, allelesToUse);
```

Comment:
This seems overly complicated.  Why does it compute a bitset and then use the bitset to determine the indeces somewhere else when it could just determine the indecies directly without the bit set?